### PR TITLE
Validation notest

### PR DESCRIPTION
**Pull request by sylvestor88**
discoverTest method implementation

### CODE REVIEW

**Comment by ptnapoleon** [comment]
The original header has to be maintained, to support junit's Parameterized runner API. So you need to do `Collection<Object[]>`, not `Collection<Object>`. Also, you left in a commented out line of debug code, please remove that

File: test/validation/org/apache/cassandra/HarnessTester.java [hunk 1]
```diff
@@ -53,11 +54,25 @@
...
+            File file = listOfFiles[i];
+            if (file.isFile() && file.getName().endsWith(".yaml"))
+            {
+                String content = FileUtils.getCanonicalPath(file);
+
+                list.add(content);
+            }
+        }
+
+        return Arrays.asList(list);
```
Comment (ptnapoleon) [inline test/validation/org/apache/cassandra/HarnessTester.java]: This is already a List, why do you need to use `Arrays.asList` here?

**Comment by ptnapoleon** [comment]
This patch doesn't compile, -1

File: test/validation/org/apache/cassandra/HarnessTester.java [hunk 2]
```diff
@@ -24,10 +24,12 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
+import java.lang.Throwable;
```
Comment (ptnapoleon) [inline test/validation/org/apache/cassandra/HarnessTester.java]: This import is unused, remove it.

**Comment by ptnapoleon** [comment]
Why have the intermediate step of adding the file names to a list, before putting them in the final result? Why not just put them straight into the `Collection<Object[]>`?

File: test/validation/org/apache/cassandra/HarnessTester.java [hunk 3]
```diff
@@ -24,10 +24,12 @@
...
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.Arrays;
```
Comment (ptnapoleon) [inline test/validation/org/apache/cassandra/HarnessTester.java]: This is also now unused.

Comment (sylvestor88) [inline test/validation/org/apache/cassandra/HarnessTester.java]: I tried putting file names directly into Collection<Object[]> yesterday but that wasn't working. I tried it now and seems to be working. I might be missing something yesterday.

**Pull request merged**